lib/core/src/chain_swap.rs

```diff
@@ -211,6 +212,11 @@ impl ChainSwapHandler {
                         .update_chain_swap_accept_zero_conf(&id, !zero_conf_rejected)?;
                 }
                 if let Some(transaction) = update.transaction.clone() {
+                    let actual_payer_amount =
+                        self.get_actual_payer_amount_from_swap_update(swap, &transaction)?;
```

Comment:
Should we query the blockchain to fetch the tx instead of using what the swapper has sent?

Reply:
My idea was to avoid it because it would delay the appearance of the pending payment until we see the tx in the mempool. Also, I thought the recoverer would catch any incongruency. After reflecting, I realize we should query the tx from the blockchain; otherwise, we risk verifying the server lockup tx as correct without double-checking the swapper-provided data.

Reply:
Right, I don't see a way we can avoid fetching the tx. I wouldn't worry about the small delay for showing it as pending as this is the receive via btc address which is by its nature a long process that waits for the transaction to confirm.

Reply:
Still have to fix a test, but here is the update in order to get the amount from the blockchain: [7ec8b52](https://github.com/breez/breez-sdk-liquid/pull/667/commits/7ec8b52fbd2d61aa4baafe5a6f0ce5f1e348bbce)